### ISSUE TITLE
Is PARSER_BEGIN/PARSER_END construct still supported?

### ISSUE THREAD

**Issue by nchamberlain** [javacc21/javacc21#41]
When I try to use the old-style PARSER_BEGIN(id)....PARSER_END(id) structure, the latest javacc-full.jar (2020-12-17) errors out with this message:
Encountered an error on (or somewhere around) line 78, column 1 of SPL.javacc
Was expecting one of the following:
_PARSER_END
Found string "PARSER_END" of type IDENTIFIER
        at com.javacc.parser.JavaCCParser.handleUnexpectedTokenType(JavaCCParser.java:28573)
        <other lines removed>
I ran   ant test    and everything worked OK. But when I searched the legacy code, I didn't find PARSER_BEGIN/PARSER_END in any of the code. So I changed the SPL.javacc to use PARSER_BEGIN/PARSER_END and got the errors above.
It seems the last commit that doesn't complain about PARSER_BEGIN/PARSER_END was committed Nov 21 with the id 384c0e3fede276c4b0f1f8a922662788658cdacb. The commit 216d2cb0bf46031bd572016aed41b611dce4ba47 on the same date fails with the above error. This version has a lot of "(token_source.inJavaCode)" statements removed, if that helps.
I haven't used github much (or javacc21, for that matter) so if I'm doing something stupid, then just tell me and then close this issue.

**Comment by revusky** [comment]
Hi Nelson. Thanks for the information! To answer the ion up top: Yes, the PARSER_BEGIN...PARSER_END is still supported. At least that is the intention. Unfortunately, it now seems that *de fact* it has not been supported for over a month. Drat! I guess I really should have left at least one of those old examples from the legacy distro with that construct so I'd have a regression test.

Try this. In the file src/main/grammars/JavaCC.javacc comment out line 373 and rebuild/retest. (That should do the trick but I haven't even checked it!) The problem was that PARSER_END is being munged (incorrectly) into an IDENTIFIER in that spot and it shouldn't be. In fact, the info is right in the error message.

As for your usage of github, it seems like you're doing everything right! Though, actually, I myself am not really a veteran Github user. Truth told, I used it for the first time about a year ago. This was because of my decade-long absence from open source activity. My Github account is actually about 12 years old, I think, but I created the account back in 2008 or so and never really used it. Instead, when I created the FreeCC project, I used this thing called GoogleCode instead. Ancient history... Well, if there's a point to that, it's that I'm actually not such an expert user of github.com (or git) myself. I still find myself "discovering" new features every now and then.
